### PR TITLE
Add forge

### PR DESCRIPTION
**Pull request by iamchrissmith**
adding Forge and dapp test without optimizations to D3M so we can run tests faster as we add to the repo

### CODE REVIEW

File: test-forge.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+if [[ -z "$MATCH" && -z "$BLOCK" ]]; then
+    forge test --fork-url "$ETH_RPC_URL" --force
+elif [[ -z "$BLOCK" ]]; then
+    forge test --fork-url "$ETH_RPC_URL" --match "$MATCH" -vvv --force
+elif [[ -z "$MATCH" ]]; then
+    forge test --fork-url "$ETH_RPC_URL" --fork-block-number "$BLOCK" --force
+else
+    forge test --fork-url "$ETH_RPC_URL" --match "$MATCH" --fork-block-number "$BLOCK" -vvv --force
+fi
```
Comment (naszam) [inline test-forge.sh:26]: eof missing newline

File: test.sh [hunk 2]
```diff
@@ -3,11 +3,23 @@ set -e
...
+
+export DAPP_BUILD_OPTIMIZE="$OPTIMIZER"
 export DAPP_BUILD_OPTIMIZE_RUNS=200
 
 if [[ -z "$1" ]]; then
   dapp --use solc:0.6.12 test --rpc-url="$ETH_RPC_URL" -v
 else
-  dapp --use solc:0.6.12 test --rpc-url="$ETH_RPC_URL" --match "$1" -vv
+  dapp --use solc:0.6.12 test --rpc-url="$ETH_RPC_URL" --match "$MATCH" -vv
 fi
```
Comment (naszam) [inline test.sh:25]: same as above

File: Makefile [hunk 3]
```diff
@@ -1,4 +1,6 @@
-all    :; DAPP_BUILD_OPTIMIZE=1 DAPP_BUILD_OPTIMIZE_RUNS=200 dapp --use solc:0.6.12 build
-clean  :; dapp clean
-test   :; ./test.sh $(match)
-deploy :; make && dapp create DssDirectDepositAaveDai 0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F 0x4449524543542d4141564556322d444149000000000000000000000000000000 0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9 0xd784927Ff2f95ba542BfC824c8a8a98F3495f6b5
+all    		 :; DAPP_BUILD_OPTIMIZE=1 DAPP_BUILD_OPTIMIZE_RUNS=200 dapp --use solc:0.6.12 build
+clean  		 :; dapp clean
+test   		 :; ./test.sh match=$(match) optimizer=1
+test-dev   	 :; ./test.sh match=$(match) optimizer=0
+test-forge   :; ./test-forge.sh match=$(match) block=$(block)
```
Comment (naszam) [inline Makefile:5]:
```suggestion
test-forge       :; ./test-forge.sh match=$(match) block=$(block)
```

**Review by naszam — COMMENTED** [review]
lgtm, just some cleanup.

File: test-forge.sh (hunk 1 above)
Comment (iamchrissmith) [reply test-forge.sh:26]: 😡 one of these days I won't make you make these comments

File: test-forge.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            *)   
+    esac
+done
+
+if [[ -z "$MATCH" && -z "$BLOCK" ]]; then
+    forge test --fork-url "$ETH_RPC_URL" --force
+elif [[ -z "$BLOCK" ]]; then
+    forge test --fork-url "$ETH_RPC_URL" --match "$MATCH" -vvv --force
+elif [[ -z "$MATCH" ]]; then
+    forge test --fork-url "$ETH_RPC_URL" --fork-block-number "$BLOCK" --force
```
Comment (naszam) [inline test-forge.sh:23]: nit: might be worth adding a bit of verbosity by default, to get some logs, for quick debugging.
```suggestion
    forge test --fork-url "$ETH_RPC_URL" --vv --force
elif [[ -z "$BLOCK" ]]; then
    forge test --fork-url "$ETH_RPC_URL" --match "$MATCH" -vvv --force
elif [[ -z "$MATCH" ]]; then
    forge test --fork-url "$ETH_RPC_URL" --fork-block-number "$BLOCK" --vv --force
```

**Review by hexonaut — APPROVED** [review]

**Review by naszam — APPROVED** [review]
LGTM

**Pull request merged**